geometry/optimization/graph_of_convex_sets.cc

```diff
@@ -167,6 +177,46 @@ std::unordered_set<Edge*> GraphOfConvexSets::Edges() {
   return edges;
 }
 
+// TODO(russt): We could get fancy and dim the color of the nodes/edges
+// according to phi, using e.g. https://graphviz.org/docs/attr-types/color/ .
+std::string GraphOfConvexSets::GetGraphvizString(
+    const std::optional<solvers::MathematicalProgramResult>& result,
+    bool show_slacks, int precision, bool scientific) const {
+  // Note: We use stringstream instead of fmt in order to control the
+  // formatting of the Eigen output and double output in a consistent way.
+  std::stringstream graphviz;
+  graphviz.precision(precision);
+  if (!scientific) graphviz << std::fixed;
+  graphviz << "digraph GraphOfConvexSets {\n";
+  graphviz << "labelloc=t;\n";
+  for (const auto& vpair : vertices_) {
+    graphviz << "v" << vpair.first << " [label=\"" << vpair.second->name();
+    if (result) {
+      graphviz << "\n x = "
+               << result->GetSolution(vpair.second->x()).transpose();
+    }
+    graphviz << "\"]\n";
+  }
+  for (const auto& epair : edges_) {
+    const Edge& e = *epair.second.get();
+    graphviz << "v" << e.u().id() << " -> v" << e.v().id();
+    graphviz << " [label=\"" << e.name();
+    if (result) {
+      graphviz << "\n";
+      graphviz << "cost = " << e.GetSolutionCost(*result);
+      if (show_slacks) {
+        graphviz << ",\n";
+        graphviz << "ϕ = " << result->GetSolution(e.phi())<< ",\n";
+        graphviz << "xu = " << e.GetSolutionXu(*result).transpose() << ",\n";
```

Comment:
Similar to Mark's point: for fractional flows this might be a little confusing (`xu` and `xv` might be nonzero and differ from the points in the sets). But I guess we decided to use `xu` and `xv` instead of `ye` and `ze` to prevent the user from being exposed to the underlying math. So I agree that printing `ye` and `ze` in the graphviz is not ideal. Another option would be use `ϕ xu = ...` and `ϕ xv = ...`. Mathematically this would be "more correct"?

Also here I'd be nice to have the vector notation as above.